packages/flutter/test/material/dropdown_test.dart

```diff
@@ -3122,4 +3122,156 @@ void main() {
         ..rrect(rrect: const RRect.fromLTRBXY(0.0, 0.0, 112.0, 47.0, 2.0, 2.0), color: Colors.grey[50], hasMaskFilter: false)
     );
   });
+
+  testWidgets('Tapping a disabled item should not close DropdownButton', (WidgetTester tester) async {
+    String? value = 'first';
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) => DropdownButton<String>(
+              value: value,
+              items: const <DropdownMenuItem<String>>[
+                DropdownMenuItem<String>(
+                  enabled: false,
+                  child: Text('disabled'),
+                ),
+                DropdownMenuItem<String>(
+                  value: 'first',
+                  child: Text('first'),
+                ),
+                DropdownMenuItem<String>(
+                  value: 'second',
+                  child: Text('second'),
+                ),
+              ],
+              onChanged: (String? v) => setState(() => value = v),
```

Comment:
1) Avoid using single letter variable names. Prefer to describe what the variables are
2) The fat arrow function (=>) implies that the value is being returned. This could be confusing at first glance since the return value does not matter here (void callback)
```suggestion
              onChanged: (String? newValue) {
                setState(() {
                  value = newValue;
                });
              },
```